lib/ansible/modules/cloud/spotinst/spotinst_event_subscription.py

```diff
@@ -0,0 +1,281 @@
+    event_subscription = spotinst.spotinst_event_subscription.Subscription()
+
+    resource_id = module.params.get('resource_id')
+    protocol = module.params.get('protocol')
+    endpoint = module.params.get('endpoint')
+    event_type = module.params.get('event_type')
+    event_format = module.params.get('event_format')
+
+    if resource_id is not None:
+        event_subscription.resource_id = resource_id
+
+    if protocol is not None:
+        event_subscription.protocol = protocol
+
+    if endpoint is not None:
+        event_subscription.endpoint = endpoint
+
+    if event_type is not None:
+        event_subscription.event_type = event_type
+
+    if event_format is not None:
+        event_subscription.event_format = event_format
+
+    return event_subscription
+# endregion
+
+
+# region Util Functions
+def handle_subscription(client, module):
+    subscription_id = None
+    message = None
+    has_changed = False
+
+    request_type, subscription_id = get_request_type_and_id(client=client, module=module)
+
+    if request_type == "create":
+        subscription_id, message, has_changed = handle_create(client=client, module=module)
+    elif request_type == "update":
+        subscription_id, message, has_changed = handle_update(client=client, module=module, subscription_id=subscription_id)
+    elif request_type == "delete":
+        subscription_id, message, has_changed = handle_delete(client=client, module=module, subscription_id=subscription_id)
+    else:
+        module.fail_json(msg="Action Not Allowed")
+
+    return subscription_id, message, has_changed
+
+
+def get_request_type_and_id(client, module):
+    request_type = None
+    subscription_id = module.params.get('id')
+    state = module.params.get('state')
+
+    if state == 'present':
+        if subscription_id is None:
+            request_type = "create"
+
+        else:
+            request_type = "update"
+
+    elif state == 'absent':
+        request_type = "delete"
+
+    return request_type, subscription_id
+
+
+def get_client(module):
+    # Retrieve creds file variables
+    creds_file_loaded_vars = dict()
+
+    credentials_path = module.params.get('credentials_path')
+
+    if credentials_path is not None:
+        try:
+            with open(credentials_path, "r") as creds:
+                for line in creds:
+                    eq_index = line.find('=')
+                    var_name = line[:eq_index].strip()
+                    string_value = line[eq_index + 1:].strip()
+                    creds_file_loaded_vars[var_name] = string_value
+        except IOError:
+            pass
+    # End of creds file retrieval
+
+    token = module.params.get('token')
+    if not token:
+        token = os.environ.get('SPOTINST_TOKEN')
+    if not token:
+        token = creds_file_loaded_vars.get("token")
+
+    account = module.params.get('account_id')
+    if not account:
+        account = os.environ.get('SPOTINST_ACCOUNT_ID') or os.environ.get('ACCOUNT')
```

Comment:
No need for this if you are using `env_fallback` mentioned in below comment.

Reply:
added in!